### ISSUE TITLE
[Bug]: Any markdown code block generated by Wave AI Chat client breaks the formatting of the chat following the code block.

### ISSUE THREAD

**Issue by xbais** [wavetermdev/waveterm#1147]
### Current Behavior

Seems like the chats are generated as markdown which is then rendered. As a result, if the generated markdown code for the chat itself contains a markdown code block, the block breaks the formatting of the chat content following the markdown code block.

### Expected Behavior

The markdown code block shouldn't affect the formatting of the rest of the generated chat.

### Steps To Reproduce

1. Open Wave AI chat window in Waveterm
2. Type this prompt : "Give me the code for a sample markdown document."
3. Analyse the formatting of the generated chat specially at the end

### Wave Version

v0.8.13

### OS

Ubuntu 24.04.1

### Architecture

x64

### Anything else?

Seems like the markdown-renderer function needs to be made more robust.

### Questionnaire

- [X] I'm interested in fixing this myself but don't know where to start
- [ ] I would like to fix and I have a solution
- [ ] I don't have time to fix this right now, but maybe later

**Comment by esimkowitz** [comment]
Hmm sounds like we need to escape text in the code blocks. If you're interested in trying to fix it, check out markdown.tsx. We use Remark as our Markdown renderer. They should have plugins that can extract the code blocks. From there, you can replace the ` character with an html-escaped equivalent.

**Comment by xbais** [comment]
Cool ! Will give it a try...

**Comment by xbais** [comment]
Cool ! Will give it a try...

**Comment by xbais** [comment]
Cool ! Will give it a try...

**Comment by xbais** [comment]
Hey @esimkowitz, I have a small doubt. the [package.json](https://github.com/wavetermdev/waveterm/blob/main/package.json) file specifies `"packageManager": "yarn@4.4.1"` , but when I try to install yarn v4.4.1, I get the error : `npm error notarget a package version that doesn't exist.`

Seems like I am missing some small detail...

**Comment by esimkowitz** [comment]
@xbais check out BUILD.md, one of the prerequisite steps is to enable corepack. We use Yarn Modern, which is distributed like a package so that the API is properly consistent for all users of our repository

**Comment by xbais** [comment]
Hey thanks for the tip! It helped me build the app using the command `task electron:start `, but when the command starts the electron app, it throws the following error:
![image](https://github.com/user-attachments/assets/0bb3f64e-0ca3-4f10-bc78-5880167724be)

Interestingly, this error : `[130375:1030/225138.119125:FATAL:credentials.cc(131)] Check failed: . : Permission denied (13)` is almost the same as the one in #1049 .

Any ideas / tips?

**Comment by esimkowitz** [comment]
Hmm, did you install the rest of the prerequisites from the BUILD.md file? You may also need to re-login for things to take effect

**Comment by xbais** [comment]
Yes, I followed all the steps in BUILD.md (for Linux) including : installation of zip, snapd and zip ... then installed NodeJS 20, checked `npm` version, did `npm install -g corepack` followed by  `corepack enable`, and then I ran the `task electron:start`.

**Comment by xbais** [comment]
This command also terminates with the same error...
![image](https://github.com/user-attachments/assets/b4b9ec9e-551b-451d-912a-ae01df3d58ae)

**Comment by xbais** [comment]
I also tried running the same command as root, assuming it could resolve some permission errors by brute-force. I used the command : `sudo /home/$USER/.nvm/versions/node/v20.18.0/bin/task electron:dev`.

Running this, returned the error : `[1031/220116.055499:FATAL:electron_main_delegate.cc(287)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.`

Error logs : 
```bash
.
.
dist/main/index.js                    1,634.48 kB
✓ built in 3.50s

build the electron main process successfully

-----

vite v5.4.10 building SSR bundle for development...
✓ 2 modules transformed.
dist/preload/preload-webview.cjs  0.66 kB │ map: 0.12 kB
dist/preload/index.cjs            3.76 kB │ map: 0.14 kB
✓ built in 13ms

build the electron preload files successfully

-----

The electron vite renderer config build.target is not "chrome?" or "es?". This could be a mistake.
[vite-plugin-static-copy] Collected 14 items.
dev server running for the electron renderer process at:

  ➜  Local:   http://localhost:5173/
  ➜  Network: use --host to expose

start electron app...

[1031/220116.055499:FATAL:electron_main_delegate.cc(287)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.
```

But now could you point me where I can add the `--no-sandbox` flag to run it this way? ... or should I leave this experiment and try something else ?

**Comment by esimkowitz** [comment]
Hmm yeah I don't think sudo is the right solution here. Sorry I haven't been as responsive, we were at GitHub Universe last week and then I moved this weekend. Will try troubleshooting this week.

**Comment by xbais** [comment]
Sure! I understand, hope you had good time at Github universe and are settled.. Do let me know if I can be of any help with the issue... 🙂

**Comment by esimkowitz** [comment]
Sorry for the delay @xbais, can you try this for me? https://sentry.io/answers/how-do-i-fix-the-error-vite-permission-denied-when-running-vite-js/#:~:text=This%20error%20occurs%20because%20your,permission%20to%20access%20the%20folder.
